concepts/80 Troubleshooting/05 DataGrid and TreeList Issues/10 Gray Boxes Appear After Data Loading/25 Eliminate Side Effects.md

```diff
@@ -0,0 +1,73 @@
+
+        <!-- tab: app.component.ts -->
+        export class AppComponent {
+            // ...
+            onContentVisibilityChange(visible) {
+                if (!visible) {
+                    this.gridScrollPosition = this.dataGrid.instance.getScrollable().scrollTop();
+                } else if (this.gridScrollPosition > 0) {
+                    this.dataGrid.instance.getScrollable().scrollTo({ top: this.gridScrollPosition }); 
+            }
+        }
+    
+    Alternatively, you can use CSS to hide the component. In this case, call the [updateDimensions](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Methods/#updateDimensions) method.
+
+##### Vue
+
+- Conditional rendering does not reload data. If you use conditional rendering to display DataGrid or TreeList, the component's scrollable container resets its scroll position while hiding the component and does not restore it while showing the component. After the component has been hidden and shown, the scroll position is reset to the first DataSource page. Once the data of the first page is not loaded, the gray boxes are displayed. Save the scroll position on hiding the component and restore it on showing so that the component displays the loaded page.
+
+        <!-- tab: App.vue -->
+        <script>
+            export default {
+                // ...
+                methods: {
+                    onContentVisibilityChange(visible) {
+                        if (!visible) {
+                            this.gridScrollPosition = this.dataGrid.instance.getScrollable().scrollTop();
+                        } else if (this.gridScrollPosition > 0) {
+                            this.dataGrid.instance.getScrollable().scrollTo({ top: this.gridScrollPosition }); 
+                        }
+                    }
+                }
+            }
+        </script>
+    
+    Alternatively, you can use CSS to hide the component. In this case, call the [updateDimensions](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Methods/#updateDimensions) method.
+
+##### React
+
+- In some cases, row placeholders appear due to unnecessary rerendering. To learn more about such issues, refer to the following guide: [Optimize Performance](/Documentation/Guide/React_Components/Optimize_Performance).
+
+- Conditional rendering does not reload data. If you use conditional rendering to display DataGrid or TreeList, the component's scrollable container resets its scroll position while hiding the component and does not restore it while showing the component. After the component has been hidden and shown, the scroll position is reset to the first DataSource page. Once the data of the first page is not loaded, the gray boxes are displayed. Save the scroll position on hiding the component and restore it on showing so that the component displays the loaded page.
```

Comment:
Again, a very complicated and confused paragraph. I do not understand anything. Please rewrite.